### PR TITLE
Use results of kube-controller-manager leader election in addon manager

### PR DESCRIPTION
**Pull request by x13n**
**What this PR does / why we need it**:
This adds leader election-like mechanism to addon manager. Currently, in a multi-master setup, upgrading one master will trigger a fight between addon managers on different masters, each forcing its own versions of addons. This leads to pod unavailability until all masters are upgraded to new version.

To avoid implementing leader election in bash, results of leader election in kube-controller-manager are used. Long term, addon manager probably should be rewritten in a real prgramming language (probably Go), and then, real leader election should be implemented there.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
I don't think there was an issue for this specifically, but this PR is related to https://github.com/kubernetes/kubernetes/issues/473

**Special notes for your reviewer**:

**Release note**:
```release-note
Addon manager supports HA masters.
```

### CODE REVIEW

File: cluster/addons/addon-manager/kube-addons.sh [hunk 1]
```diff
@@ -143,6 +148,23 @@ function ensure_addons() {
   log INFO "== Kubernetes addon ensure completed at $(date -Is) =="
 }
 
+function is_leader() {
```
Comment (mikedanese) [inline cluster/addons/addon-manager/kube-addons.sh]: This real pokes around at the internals. We want to start moving the addon manager to use the front door. Can you look at the controller-manager leader election result instead?

Comment (mikedanese) [reply cluster/addons/addon-manager/kube-addons.sh]: The leader election results for kube-controller-manager are stored on it's endpoints object:

kubectl -n kube-system get ep kube-controller-manager -o json

Comment (x13n) [reply cluster/addons/addon-manager/kube-addons.sh]: Thanks for pointing this one! Done.

**Comment by x13n** [comment]
Looks like there is a regression introduced by https://github.com/kubernetes/kubernetes/pull/53440

**Comment by x13n** [comment]
Created https://github.com/kubernetes/kubernetes/pull/55694 to get rid of the regression.

**Pull request merged**